tests/drivers/abp2/main.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2024 CNRS, France
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the Honeywell ABP2 series
+ *              pressure and temperature sensor driver.
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "ztimer.h"
+#include "timex.h"
+#include "phydat.h"
+#include "abp2.h"
+#include "abp2_params.h"
+#include "saul_reg.h"
+
+#define STATUS_SLEEP_MS         (1)
+#define MEASUREMENT_SLEEP_MS    (1000)
+#define MAX_LOOPS_STATUS        (10)
+#define MAX_LOOPS_MEAS          (3)
+
+/* supported acquisition modes */
+enum {
+    ABP2_NON_BLOCKING,
+    ABP2_BLOCKING,
+    ABP2_SAUL
+} acqmode = ABP2_NON_BLOCKING;
+
+static abp2_t dev;
+
+int main(void)
+{
+    int32_t press = 0;
+    int32_t temp = 0;
+    abp2_params_t prms;
+    phydat_t phyPress;
+    phydat_t phyTemp;
+    void *ptr;
+    int cntMeas = 0;
+    saul_reg_t *saulDev = saul_reg;
+    phydat_t saulData;
+
+    puts("ABP2 Honeywell series pressure and temperature sensor\n");
+
+    if (saulDev == NULL) {
+        puts("ERROR: no ABP2_SAUL devices detected");
+    }
+
+    ptr = memcpy(&prms, &abp2_params, sizeof(abp2_params));
+    if (!ptr)
+    {
+       puts("main() >> ERROR: memcpy() failed");
+       return 1;
+    }
+
+    printf("Initializing SPI bus %d\n", prms.spi);
+    spi_init(prms.spi);
+
+    printf("Initializing ABP2 at SPI_DEV(%i)... ", prms.spi);
+
+    if (abp2_init(&dev, &prms) == 0)
+    {
+        puts("[OK]");
+    }
+    else {
+        puts("[Failed]");
+        return -1;
+    }
+
+    puts("=========================");
+    puts("        Measuring");
+    puts("=========================");
+
+    printf("Pressure range = %d .. %d\n", (int)prms.rangeMin, (int)prms.rangeMax);
+
+    while (1) {
+        int res;
+
+        ztimer_sleep(ZTIMER_MSEC, MEASUREMENT_SLEEP_MS);
```

Comment:
```suggestion
        ztimer_sleep(ZTIMER_USEC, MEASUREMENT_SLEEP_MS * US_PER_MS);
```

or  in `<test>/Makefile`

```
USEMODULE += ztimer_msec 
```